addons/web/static/src/views/calendar/calendar_model.js

```diff
@@ -732,13 +732,20 @@ export class CalendarModel extends Model {
             type: "dynamic",
             recordId: null,
             value,
-            label: formatter(rawValue, { field }) || _t("Undefined"),
+            label: formatter(rawValue, { field }) || this.defaultFilterLabel(),
             active: previousFilter ? previousFilter.active : true,
             canRemove: false,
             colorIndex,
             hasAvatar: !!value,
         };
     }
+    /**
+     * @protected
+     * @returns {string}
+     */
+    defaultFilterLabel() {
```

Comment:
It could be a getter and can you move it with the others getters. 🙂 
```suggestion
    get defaultFilterLabel() {
```